pkg/hub/hubclient/hubclient.go

```diff
@@ -143,25 +146,33 @@ func PostClusterInfoToHub(ctx context.Context, spokeclient client.Client, hubCli
 }
 
 func PostDashboardCredsToHub(ctx context.Context, spokeclient, hubClient client.Client) error {
-	sa := &corev1.ServiceAccount{
+	tokenReq := &authv1.TokenRequest{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      KubeSliceDashboardSA,
 			Namespace: controllers.ControlPlaneNamespace,
 		},
+		Spec: authv1.TokenRequestSpec{
+			// TODO: check if we need to add any parameters here
+		},
 	}
-	if err := spokeclient.Get(ctx, types.NamespacedName{Name: sa.Name, Namespace: controllers.ControlPlaneNamespace}, sa); err != nil {
-		log.Error(err, "Error getting service account")
+
+	if err := spokeclient.Create(ctx, tokenReq); err != nil {
+		log.Error(err, "Error getting service account token")
 		return err
 	}
 
-	secret := &corev1.Secret{}
-	err := spokeclient.Get(ctx, types.NamespacedName{Name: sa.Secrets[0].Name, Namespace: controllers.ControlPlaneNamespace}, secret)
+	ca, err := ioutil.ReadFile(CA_FILE)
```

Comment:
We are creating a token for the dashboard's service account and using the certs from the worker operator's pod. Can we instead look for a Secret that is bound to the dashboard's service account and read the token and the certs from it? Or create a Secret (i am assuming a token and certs are generated automatically if you create a Secret) and bind it to the service account and read the certs from it?

Reply:
1. The CA cert is common for the cluster, so it doesn't matter where we read it from.
2. From 1.24, secrets are not automatically created for serviceaccounts, which is why I switched to tokenrequest api. Creating a secret is unnecessary since the only purpose of the secret is to fetch the token once.